modules/core/src/main/kotlin/com/github/l34130/mise/core/wsl/WslPathUtils.kt

```diff
@@ -0,0 +1,531 @@
+        val pathWithoutDrive = driveMatch.groupValues[2]
+
+        // Normalize prefix (remove trailing slashes)
+        // For root "/" this becomes empty string "", which is correct: "" + "/" + "c" = "/c"
+        val normalizedPrefix = mountPrefix.trimEnd('/')
+
+        // Convert to WSL mount path
+        val unixPath = pathWithoutDrive.replace('\\', '/')
+        return "$normalizedPrefix/$driveLetter/$unixPath"
+    }
+
+    /**
+     * Converts a Windows path to a WSL mount path.
+     *
+     * Examples (with default /mnt prefix):
+     * - "C:\Users\project" → "/mnt/c/Users/project"
+     * - "D:\workspace" → "/mnt/d/workspace"
+     *
+     * Examples (with custom / prefix):
+     * - "C:\Users\project" → "/c/Users/project"
+     *
+     * @param windowsPath The Windows path to convert
+     * @return The converted WSL mount path using configured mount prefix
+     * @throws IllegalArgumentException if windowsPath format is invalid
+     */
+    fun convertWindowsToWslMountPath(windowsPath: String): String {
+        // Allow conversion on Windows or when running inside WSL
+        if (!SystemInfo.isWindows && !isRunningInWsl()) return windowsPath
+
+        // Get configured mount prefix, default to /mnt if not configured
+        val mountPrefix = readWslMountPrefix() ?: "/mnt"
+
+        return convertWindowsToWslMountPathWithPrefix(windowsPath, mountPrefix)
+    }
+
+    /**
+     * Validates that a UNC path exists and is accessible.
+     *
+     * @param uncPath The UNC path to validate
+     * @return true if the path exists and is accessible, false otherwise
+     */
+    fun validateUncPath(uncPath: String): Boolean {
+        // When not on Windows or WSL, skip validation (always return true)
+        if (!SystemInfo.isWindows && !isRunningInWsl()) return true
+        if (uncPath.isBlank()) return false
+
+        return try {
+            val path = Paths.get(uncPath)
+            Files.exists(path)
+        } catch (e: Exception) {
+            logger.warn("Failed to validate UNC path: $uncPath", e)
+            false
+        }
+    }
+
+    /**
+     * Converts a mise tool's install path for WSL compatibility.
+     *
+     * On non-Windows systems or when WSL mode is disabled, returns the original path unchanged.
+     * On Windows with WSL mode enabled, converts the Unix path to a Windows UNC path.
+     *
+     * @param tool The MiseDevTool containing the install path to convert
+     * @return The converted path (Windows UNC path for WSL, original path otherwise)
+     * @throws IllegalStateException if WSL mode is enabled but distribution is not configured,
+     *         or if the converted UNC path is not accessible
+     */
+    fun convertToolPathForWsl(tool: MiseDevTool): String {
+        if (!SystemInfo.isWindows) return tool.installPath
+
+        val settings = application.service<MiseApplicationSettings>()
+        if (!settings.state.isWslMode) return tool.installPath
```

Comment:
Sorry, the PR conflicts now.

Please update the `tool.installPath` with `tool.shimsInstallPath()`